balancer/rls/control_channel.go

```diff
@@ -57,24 +57,22 @@ type controlChannel struct {
 	// hammering the RLS service while it is overloaded or down.
 	throttler adaptiveThrottler
 
-	cc                  *grpc.ClientConn
-	client              rlsgrpc.RouteLookupServiceClient
-	logger              *internalgrpclog.PrefixLogger
-	connectivityStateCh *buffer.Unbounded
-	unsubscribe         func()
-	monitorDoneCh       chan struct{}
+	cc                   *grpc.ClientConn
+	client               rlsgrpc.RouteLookupServiceClient
+	logger               *internalgrpclog.PrefixLogger
+	unsubscribe          func()
+	seenTransientFailure bool
+	mu                   sync.Mutex
```

Comment:
Nit: Please group the mutex and the fields that it guards as a separate block. It is a generally used convention that any fields that come right below a mutex are to be guarded by that mutex.